File: python/device.py
class Device:
    def __init__(self, my_name):
        self.name = my_name
        self.status = ''

    def set_status(self, new_status):
        self.status = new_status

    def get_status(self):
        return self.status


class DeviceTest:
    def test_KitchenLightOn(self):
        device = Device('kitchen light')
        device.set_status('on')
        return device.get_status() == 'on'


myTestSuite = DeviceTest()
myTestResult = myTestSuite.test_KitchenLightOn()
print('passed' if myTestResult else 'failed')

File: ts/device.ts
class Device {
    name: string;
    status: string;

    constructor(my_name: string) {
        this.name = my_name;
        this.status = '';
    }

    setStatus(new_status: string): void {
        this.status = new_status;
    }

    getStatus(): string {
        return this.status;
    }
}

class DeviceTest {
    test_KitchenLightOn(): boolean {
        let device = new Device('kitchen light');

        device.setStatus('on');

        let myResult = device.getStatus() === 'on'

        return myResult;
    }
}

let myTestSuite = new DeviceTest();
let myTestResult = myTestSuite.test_KitchenLightOn();

console.log(myTestResult ? 'passed' : 'failed');
